### ISSUE TITLE
Disabled branch protection

### ISSUE THREAD

**Issue by cromerc** [cromnix/packages-cromnix#9]
Considering the ammount of changes you will be doing to this github repo I disabled the master branch protection. You can now merge all changes without having to wait for me.

**Comment by udeved** [comment]
I think I found a good way to sync from arch PKGBUILDs that don't eed to be altered, eg all the dev stuff.
For example qt, python, perl modules could be auto imported into their branches, and merged from there in world branch.
I can also create lists of other builds that are untouched to reduce maintenance
So atomizing the imports into smaller groups(branches) seems to work well so far.

Example:

arch bump qt5 builds, we would run import script in the branch, and then merge it in world later. We could also add additional eg system-testing and world-testing branches to merge fresh arch buids in there before they go in stable system and world branches.

**Comment by udeved** [comment]
This approach could eventually compensate for small team along with buildbot.
We would end up with arch import lists, and cromnix black lists, and the script would be added to artools.

```
system
|--- cromnix (our packages)
|--- arch import branch(es)
world
|--- cromnix (our packages)
|--- arch import branch(es)
        |---- qt
        |---- python
        |---- perl module
```

Generic workflow.

**Comment by udeved** [comment]
Statistics:

system:  204 builds
world: 513 builds

**Comment by aadityabagga** [comment]
@udeved its a nice idea in my opinion, could reduce the workload. It should work with the buildbot since the buildbot works based on changes in PKGBUILD, whether they be manual changes or changes due to import script. It could fail when number of packages changed in a single pull request / commit are large though.

**Comment by udeved** [comment]
@aadityabagga 

I guess its a matter to make the feature branches with builds in terms of time limit smaller.

**Comment by udeved** [comment]
@aadityabagga 

From you experience so far, would this be working from time limit pov?
Enough atomized to match time limit? 

https://github.com/cromnix/cromnix/pulls

**Comment by aadityabagga** [comment]
@udeved nope dude, from my experience following is the status:

large (kernel build): [failed](https://travis-ci.org/aadityabagga/packages-core/builds/243372494)
medium-large (qt5 + 1-2 medium packages): [built](https://travis-ci.org/aadityabagga/packages-openrc/builds/242918926)
medium: 5-8 packages [built](https://travis-ci.org/aadityabagga/packages-community/builds/244191891), more than that: [failed](https://travis-ci.org/aadityabagga/packages-community)
medium-small: 10-15 packages [built](https://travis-ci.org/aadityabagga/packages)
small: 20-50 packages

**Comment by cromerc** [comment]
Hmmm, so failure is still hight with medium and large packages. What else could we do?

**Comment by udeved** [comment]
Too bad.
Its probably an option to do it like arch, to have nested git repos inside a git repo, ie every single build has its own repo. Not sure if github limits repos.

**Comment by udeved** [comment]
Or, we fine tune branches, branch out to atomize it further, if necessary, one branch per build set for travis?
So we better keep master empty to branch out?

**Comment by aadityabagga** [comment]
@udeved We could initially create some packages ourselves (ie, bootstrap a repo), and then as they get updated we give them to the buildbot to build.

Current buildbot only determines package changes based on source PKGBUILDs being updated; we could add a check for existing package being built before building new package, which could save time and resources and allow building different packages. Implementation can be adapted from [here](https://github.com/arch-openrc/build-openrc/blob/master/functions_openrc#L3) and [here](https://github.com/arch-openrc/build-openrc/blob/master/build_openrc.sh#L146).

**Comment by udeved** [comment]
Would be my solution too @aadityabagga 

Clone basically core repo initially, and update. We need a solid base we can build against, and doing it with only 3 people all by ourselves may be too much to handle. We don't need to reinvent the wheel here in my view.

**Comment by cromerc** [comment]
@udeved So you would prefer to fork arch linux then if I am understanding?

**Comment by udeved** [comment]
Yes, fork to start off to better compile. I see some problems, of we basically use already in the core system our own depends. For example, I was gonna ask, if you could basically build the arch toolchain and put it in a separate repo? My i3 doesn't like the gcc heavy load, but I would take care of the rest so we have a base-devel group.
What I am saying, it is so much easier to build your system how you want it later on, if you start off with a fully featured gcc for example, and don't deviate too much from arch set depends that are not in some way tied to systemd.
What I think we should do, start with tool chain:

* gcc
* glibc
* binutils
* linux-api-headers

If we have this, we can build packages against arch packages that we don't have ourselves atm.
Really depends what you want to achieve, but if you want a desktop, you need udisks2, upower and stuff, and this is with your current collection of builds still long way to go.
The cromnix repo I created with the many branches has almost 800 packages, all stuff we need to get a desktop. @cromerc

**Comment by cromerc** [comment]
Then I would like to make a proposal. We should split the current cromnix that exists from the fork of arch that we will do. So the split would look like this:
Cromnix - "the packages the way they are now which are not a fork of arch linux"
Cromnix Vanilla - "the fork of arch linux without systemd using openrc as it's primary init system"

The reason for the split would be that for my thesis for the university I need it to not be a fork of another distro, it needs to be built from the ground up. Basically all of the OpenRC, elogind, and any nonsystemd packages would be shared between the 2 sub projects of cromnix. The vanilla branch would contain the fork of the arch project while the other branch would be a pure branch from linux from scratch.

The reason I was thinking of calling it "cromnix vanilla" is because that is what arch linux is, the packages are clean, unmodified, and do not include configuration. Plus it is aimed at more advanced users. The project I am working on for my thesis is an OpenRC distro built from scratch, but designed to be more user friendly similar to Ubuntu with point releases.

Of course the name "vanilla" is just an idea. If you have a better name, please feel free to give it a better name.

As far as forking goes, wouldn't it be better to start from arch-openrc? Basically we copy the current arch PKGBUILDS and arch-openrc PKGBUILDS. Then we copy the repos for arch linux and arch-openrc. Then it's just a matter of removing the systemd PKGBUILDS and packages and replace them with their nosystemd counterparts. Doing it this way would make forking much quicker since many of the nosystemd packages already exist and are already compiled and available in the arch-openrc SF repos. As you said, why reinvent the wheel, I think forking from arch-openrc would accomplish everything you want to accomplish but much quicker.

**Comment by cromerc** [comment]
I am getting on IRC now, if you would like to discuss it further.

**Comment by cromerc** [comment]
The download of all the binary packages finished, 38GB of packages for arch linux just for 64bit. I also ran `buildtree -as` when the sync was completed so that we have the PKGBUILDS that correspond to the binary packages that are in the repos.

**Comment by udeved** [comment]
@cromerc 

Whats the size of the core repo, and how many packages do you have in there?
You you simply do a `ls > core.list` in the folder please?

**Comment by cromerc** [comment]
The core repo is 441.4 MB

The 38GB includes extra, multilib, and community.

[core.list.txt](https://github.com/cromnix/packages-cromnix/files/1117926/core.list.txt)

**Comment by udeved** [comment]
Could you please use pastbin or so, the attached file is not txt. Github makes rubbish of it.

**Comment by cromerc** [comment]
https://pastebin.com/3Fm803X9

**Comment by udeved** [comment]
Thx, I can upload core later. ;)

**Comment by udeved** [comment]
@cromerc 

These are our 47 blacklist packages, we need to build them.

https://pastebin.com/sFu2GSy9

**Comment by udeved** [comment]
@cromerc 

Could you please delete this repo?

Regarding fork name, camnix? chris,aaditya,marcus initials

**Comment by cromerc** [comment]
Sounds good, so rename the other repo to packages-camnix?

**Comment by udeved** [comment]
Nope, I'd want it deleted, and after I have transferred the overlay branches to a new camnix repo, you could delete cromnix and packages-cromnix.

**Comment by udeved** [comment]
Or we make it artix,somewhat resembling its parent name.

**Comment by udeved** [comment]
I went artix now, better distinguishable name.

@cromerc 

Ca you please nuke cromnix and packages-cromnix repos?
